### PR TITLE
Fix possible NPE for current user

### PR DESCRIPTION
**Pull request by Beacze**

### CODE REVIEW

**Comment by henning-gerhardt** [comment]
TravisCI is failing

```
Results :
Tests in error: 
  org.kitodo.services.workflow.WorkflowControllerServiceTest
```

and there is a uncatched exception in test `org.kitodo.services.file.FileServiceTest`

```
08:20:22.538 [main] ERROR org.kitodo.services.file.FileServiceTest - Could not create directory.
java.io.IOException: Could not create directory.
	at org.kitodo.filemanagement.FileManagement.createDirectory(FileManagement.java:57) ~[kitodo-file-management-3.0-SNAPSHOT.jar:?]
	at org.kitodo.filemanagement.FileManagement.create(FileManagement.java:50) ~[kitodo-file-management-3.0-SNAPSHOT.jar:?]
	at org.kitodo.services.file.FileService.createDirectory(FileService.java:99) ~[classes/:?]
	at org.kitodo.services.file.FileServiceTest.testCreateDirectoryWithMissingRoot(FileServiceTest.java:87) [test-classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_151]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_151]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_151]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_151]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50) [junit-4.12.jar:4.12]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit-4.12.jar:4.12]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47) [junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325) [junit-4.12.jar:4.12]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78) [junit-4.12.jar:4.12]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268) [junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363) [junit-4.12.jar:4.12]
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252) [surefire-junit4-2.12.4.jar:2.12.4]
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141) [surefire-junit4-2.12.4.jar:2.12.4]
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112) [surefire-junit4-2.12.4.jar:2.12.4]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_151]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_151]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_151]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_151]
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189) [surefire-api-2.12.4.jar:2.12.4]
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165) [surefire-booter-2.12.4.jar:2.12.4]
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85) [surefire-booter-2.12.4.jar:2.12.4]
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115) [surefire-booter-2.12.4.jar:2.12.4]
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75) [surefire-booter-2.12.4.jar:2.12.4]
```

File: Kitodo/src/test/java/org/kitodo/services/workflow/WorkflowControllerServiceTest.java [hunk 1]
```diff
@@ -22,12 +22,16 @@
...
+    private static WorkflowControllerService workflowControllerService;
 
     @BeforeClass
     public static void prepareUser() {
-        User user = new User();
-        workflowControllerService.setUser(user);
+        try {
+            workflowControllerService = new ServiceManager().getWorkflowControllerService();
+        } catch (Exception e) {
+            User user = new User();
```
Comment (henning-gerhardt) [inline Kitodo/src/test/java/org/kitodo/services/workflow/WorkflowControllerServiceTest.java]: Is setting up a new empty user object correct solution in any case an exception is thrown on initialising the test suite?

Comment (henning-gerhardt) [reply Kitodo/src/test/java/org/kitodo/services/workflow/WorkflowControllerServiceTest.java]: Ok, but I see here a second issue: if an exception is thrown how did you know that `workflowControllerService` is correct initialised and usable?

Comment (Beacze) [reply Kitodo/src/test/java/org/kitodo/services/workflow/WorkflowControllerServiceTest.java]: Good point, I should change to this certain exception.

**Comment by henning-gerhardt** [comment]
There is at least one merge conflict in

- Kitodo/src/main/java/de/sub/goobi/helper/WebDav.java

Please fix this conflict.

**Pull request merged**